### ISSUE TITLE
Make namelist buffer size runtime configurable?

### ISSUE THREAD

**Issue by mnlevy1981** [marbl-ecosys/MARBL#33]
Right now we hard-code three parameters needed to pass namelists from GCM to MARBL:

```
marbl_nl_in_size is the number of characters in the entire namelist file
marbl_nl_cnt is the number of distinct namelists in the file
marbl_nl_buffer_size is the number of characters in the largest namelist
```

It would be great if there was a way to configure these values at run-time, though we need a smart way to do it. Maybe let the GCM determine them and pass values to MARBL during init?

**Comment by mnlevy1981** [comment]
One possibility would be have `init()` and `config()` determine the values from `gcm_nl_buffer`, which should have `len=marbl_nl_buffer_size` and dimension `marbl_nl_cnt`; `marbl_nl_in_size` is only used in

```
    character(len=marbl_nl_in_size), intent(in) :: str_in
```

which could easily be replaced with

```
    character(len=*), intent(in) :: str_in
```

I think at the very least this is worth investigating for MARBL 1.0.0, though it may get pushed to the post-MARBL 1.0.0 milestone.

**Comment by mnlevy1981** [comment]
Closed via #185
